test/Tests.Infrastructure/AsyncQueueExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Threading.Tasks;
+using Sparrow.Server.Collections;
+
+namespace Tests.Infrastructure;
+
+public static class AsyncQueueExtensions
+{
+    private static readonly TimeSpan DefaultTimeout = TimeSpan.FromSeconds(1);
+    
+    public static async Task<T> DequeueUntilAsync<T>(this AsyncQueue<T> queue, Predicate<T> predicate, TimeSpan? maxWaitForDequeue = null)
+    {
+        // For the first, wait asynchronously with no timeout
+        T first = await queue.DequeueAsync();
```

Comment:
not sure why

Reply:
When I was writing it, it looked smart. Now it does not. Let me simplify and retest

Reply:
The reason to wait with no timeout here is to wait for some setup that might take more than timeout. For example notifications, if the setup takes 2s and we timeout it will break the test. The first timeout-less wait ensure that there's always enough of time to set things up. 

Will add this as a comment